llvm/lib/CheerpWriter/CheerpWasmWriter.cpp

```diff
@@ -1266,7 +1266,7 @@ uint32_t CheerpWasmWriter::compileLoadStorePointer(WasmBuffer& code, const Value
 		if(const GlobalVariable* GV = dyn_cast<GlobalVariable>(p))
 			gepWriter.addConst(linearHelper.getGlobalVariableAddress(GV));
 		else if(isa<IntToPtrInst>(p) && isa<ConstantInt>(cast<IntToPtrInst>(p)->getOperand(0)))
-			gepWriter.addConst(cast<ConstantInt>(cast<IntToPtrInst>(p)->getOperand(0))->getZExtValue());
+			gepWriter.addConst(cast<ConstantInt>(cast<IntToPtrInst>(p)->getOperand(0))->getSExtValue());
```

Comment:
This looks suspicious, inttoptr should imply that the int is 32-bit wide, in which case there should be no difference between using ZExt or SExt